### PR TITLE
[RFC]topology: pre-processor: support to include conf block with IncludeByKey

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Currently, The IncludeByKey mechanism only supports to conditionally include topology conf file. Even if we only want to conditionally include a small conf block, we have to use a conf file, and in the end, we will have a lot of trivial conf files, with each only contains a small conf block.

This patch extends the use of IncludeByKey, and make it support to include conf block conditionally.

Signed-off-by: Chao Song <chao.song@linux.intel.com>

### CODE REVIEW

File: topology/pre-processor.c [hunk 1]
```diff
@@ -504,23 +507,39 @@ static int pre_process_include_conf(struct tplg_pre_processor *tplg_pp, snd_conf
...
-		else
-			full_path = tplg_snprintf("%s", filename);
 
-		ret = snd_input_stdio_open(&in, full_path, "r");
-		if (ret < 0) {
-			fprintf(stderr, "Unable to open included conf file %s\n", full_path);
+		/*
+		 * Check whether the value for the matched key is a conf file or conf block,
+		 * If it ends with '.conf', it is a conf file, otherwise, it is conf block.
+		 */
```
Comment (lgirdwood) [inline topology/pre-processor.c:519]: I would add in the comment whet the supported input strings are and how they are converted to the runtime string value.

**Review by lgirdwood — COMMENTED** [review]
@ranj063 @juimonen pls review.

File: topology/pre-processor.c [hunk 2]
```diff
@@ -485,9 +485,12 @@ static int pre_process_include_conf(struct tplg_pre_processor *tplg_pp, snd_conf
 		snd_input_t *in;
 		snd_config_t *n;
 		regex_t regex;
-		const char *filename;
```
Comment (ranj063) [inline topology/pre-processor.c:488]: @aiChaoSONG the code looks good. Can you please add an example of this in the commit message and what the limitations are?

**Comment by ranj063** [comment]
posted upsream https://github.com/alsa-project/alsa-utils/pull/187

**Comment by ranj063** [comment]
merged upstream

**Pull request closed without merging**